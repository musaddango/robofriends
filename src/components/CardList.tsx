import React from 'react';
import Card from './Card.tsx';

interface RobotObject{
    id: number,
    name: string,
    username?: string,
    email: string,
  }

const CardList = ({ robots }) => {
  return (
    <div>
      {
        robots.map((user: RobotObject, i: number) => {
          return (
            <Card
              key={i}
              id={user.id}
              name={user.name}
              email={user.email}
              />
          );
        })
      }
    </div>
  );
}

export default CardList;