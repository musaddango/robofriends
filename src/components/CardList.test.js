import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CardList from './CardList';

it('expect to render Card component', ()=>{
    const mockRobots = [
        {
            id: 1,
            username: 'musadango',
            name: 'Musa Dango',
            email: 'musa@x.com',
        }
    ]
    const wrapper = shallow(<CardList robots={mockRobots} />)

    expect(toJson(wrapper)).toMatchSnapshot();
    // shallow(<CardList robots={mockRobots} />).debug();
});