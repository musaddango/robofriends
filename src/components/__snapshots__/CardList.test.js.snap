// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div>
  <Card
    email="musa@x.com"
    id={1}
    key="0"
    name="Musa Dango"
  />
</div>
`;
